internal/pkgecosystem/download.go

```diff
@@ -0,0 +1,89 @@
+package pkgecosystem
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+)
+
+// getLastUrlElement returns the last element in a (URL) string split on the '/' character
+func getLastUrlElement(url string) string {
```

Comment:
s/Url/URL/.

Also, can we just use `path.Base` here? 

https://pkg.go.dev/path#Base

Reply:
Yes